chunker/chunk_documents_excel.py

```diff
@@ -0,0 +1,304 @@
+########################################################################################## 
+
+class DataProcessor:
+    # create a constructor to initialize the attributes of an object
+    def __init__(self, filepath):
+        self.filepath = filepath
+        self.df = None
+        
+    # function to load data from the excel file to the dataframe
+    def load_data(self):
+        # load the excel file to the df
+        self.df = pd.read_excel(self.filepath, header = [0, 1, 2])
+        
+        return self.df
+    
+    # function to clean the dataframe
+    def clean_data(self):
+        # drop empty rows and columns
+        self.df.dropna(axis = 0, how = 'all', inplace = True)
+        self.df.dropna(axis = 1, how = 'all', inplace = True)
+        
+        # reset the index
+        self.df = self.df.reset_index(drop = True)
+        
+        # drop the major section headers and reset the index
+        self.df = self.df[~((self.df.iloc[:, 0].str.isupper()) & (self.df.iloc[:, 1:].isnull().all(1)))]
+        self.df.reset_index(drop = True, inplace = True)
+        
+        return self.df
+
+##########################################################################################
+# CHUNKING FUNCTIONS
+########################################################################################## 
+
+
+
+
+class DataChunker:
+    # create a constructor to initialize the attributes of an object
+    def __init__(self, df, output_dir = 'tables', custom_folder_name = None):
+        self.df = df
+        self.output_dir = output_dir
+        
+        if custom_folder_name:
+            self.output_dir = os.path.join(self.output_dir, custom_folder_name)
+    
+        if not os.path.exists(self.output_dir):
+            os.makedirs(self.output_dir)
+
+    
+
+    # function to add section title as context to likert scale rows
+    def add_context_to_likert_scale(self, df_chunk, title):
+
+        sentiment_start = None
+        sentiment_end = None
+
+        for i, row in df_chunk.iterrows():
+            if isinstance(row[0], str):
+                if sentiment_start is None:
+                    # update the start sentiment to the row value
+                    sentiment_start = row[0].strip() # remove leading and trailing spaces to ensure there aren't unnecessary spaces when appending
+
+                # keep updating until the last row is found
+                sentiment_end = row[0].strip() # remove leading and trailing spaces to ensure there aren't unnecessary spaces when appending 
+                
+        # function to append the specific sentiment based on the number
+        def append_sentiment(row):
+            if row[0] in [1, 2]:
+                return f"{sentiment_start}_{row[0]}"
+            elif row[0] in [5, 6]:
+                return f"{sentiment_end}_{row[0]}"
+            elif row[0] in [3, 4]:
+                return f"Context Dependent_{row[0]}"
+            else:
+                return row[0] 
+        
```

Comment:
this func might break if there are any non-num values in the col. Should probably do a check first

Reply:
@egdagger